embedded/embedded-test/src/test/java/org/mule/runtime/module/embedded/AbstractEmbeddedTestCase.java

```diff
@@ -0,0 +1,161 @@
+                                     boolean validateUsageOfDeploymentService)
+      throws Exception {
+    doWithinArtifact(applicationBundleDescriptor, artifactFolder, portConsumer, lazyInitializationEnabled, xmlValidationsEnabled,
+                     log4JConfigurationFileOptional, validateUsageOfDeploymentService,
+                     APPS_FOLDER, (container, artifactConfiguration) -> container.getDeploymentService()
+                         .deployApplication(artifactConfiguration));
+  }
+
+  protected void doWithinDomain(BundleDescriptor applicationBundleDescriptor, String artifactFolder,
+                                Consumer<Integer> portConsumer)
+      throws Exception {
+    doWithinArtifact(applicationBundleDescriptor, artifactFolder, portConsumer, false, true, empty(), true,
+                     DOMAINS_FOLDER, (container, artifactConfiguration) -> container.getDeploymentService()
+                         .deployDomain(artifactConfiguration));
+  }
+
+  protected void doWithinArtifact(BundleDescriptor applicationBundleDescriptor,
+                                  String artifactFolder,
+                                  Consumer<Integer> portConsumer,
+                                  boolean lazyInitializationEnabled,
+                                  boolean xmlValidationsEnabled,
+                                  Optional<URI> log4JConfigurationFileOptional,
+                                  boolean validateUsageOfDeploymentService,
+                                  String artifactDeploymentFolder,
+                                  BiConsumer<EmbeddedContainer, ArtifactConfiguration> deployConsumer)
+      throws Exception {
+    File artifactFile = installMavenArtifact(artifactFolder, applicationBundleDescriptor);
+    Integer httpListenerPort = new FreePortFinder(6000, 9000).find();
+    testWithSystemProperty("httpPort", valueOf(httpListenerPort), () -> {
+      embeddedTestHelper.recreateContainerFolder();
+      embeddedTestHelper.testWithDefaultSettings(embeddedContainerBuilder -> {
+        try {
+          embeddedContainerBuilder.log4jConfigurationFile(log4JConfigurationFileOptional
+              .orElse(getClass().getClassLoader().getResource("log4j2-default.xml").toURI()))
+              .product(MULE)
+              .build();
+
+        } catch (Exception e) {
+          throw new RuntimeException(e);
+        }
+      }, (container) -> {
+        ArtifactConfiguration artifactConfiguration = ArtifactConfiguration.builder()
+            .artifactLocation(artifactFile)
+            .deploymentConfiguration(DeploymentConfiguration.builder()
+                .lazyInitialization(lazyInitializationEnabled)
+                .xmlValidations(xmlValidationsEnabled)
+                .build())
+            .build();
+        deployConsumer.accept(container, artifactConfiguration);
+        // TODO MULE-10392: To be removed once we have methods to deploy with properties, unify code for deployment!
```

Comment:
Remove this comment as we ended up having these configurations as part of the API. Internally the embedded code uses deployment properties to propagate this kind of deployment mode.